src/key.rs

```diff
@@ -417,6 +426,35 @@ mod test {
         assert_eq!(PublicKey::from_slice(&pk1.serialize_uncompressed()[..]), Ok(pk1));
     }
 
+    #[test]
+    fn pubkey_write_round_trip() {
+        let s = Secp256k1::new();
+
+        let (_, pk1) = s.generate_keypair(&mut thread_rng());
+
+        let mut compressed_vec = Vec::new();
+        assert!(pk1.write(&mut compressed_vec).is_ok());
+        assert_eq!(PublicKey::from_slice(&compressed_vec), Ok(pk1));
+
+        let mut compressed_buf = [0 as u8; constants::PUBLIC_KEY_SIZE];
```

Comment:
The implementation of `io::Write` for `&'a mut [u8]` says that: "Note that writing updates the slice to point to the yet unwritten part. The slice will be empty when it has been completely overwritten."

Is this good API ergonomics? As you see here users of this function are fine if they pass in a `Vec`, but in the case of a mutable borrow of arrays, special scoping is required.

Reply:
It sounds perfectly reasonable to me, but also has nothing to do with rust-secp.